svc/ctrl/proto/hydra/v1/workspace_quota_sync.proto

```
@@ -0,0 +1,41 @@
+// Singleton VO keyed "default" so overlapping triggers serialize
+// instead of double-writing.
+service WorkspaceQuotaSyncService {
+  option (dev.restate.sdk.go.service_type) = VIRTUAL_OBJECT;
+
+  // RunSync reads every row from MySQL `quota` (small table, no
+  // pagination needed) and INSERTs them into CH `workspace_quotas`.
+  // CH's ReplacingMergeTree(updated_at) collapses duplicates per
+  // workspace_id keeping the latest sync, so re-runs are idempotent
+  // and there's no TRUNCATE-window where the dict sees an empty
+  // mirror. Heartbeats on success.
+  rpc RunSync(RunWorkspaceQuotaSyncRequest) returns (RunWorkspaceQuotaSyncResponse) {}
```

Buf (1.69.0)

[error] RPC request type "RunWorkspaceQuotaSyncRequest" should be named "RunSyncRequest" or "WorkspaceQuotaSyncServiceRunSyncRequest".

(RPC_REQUEST_STANDARD_NAME)